packages/rest/src/__tests__/acceptance/caching-interceptor/global-caching-interceptor.acceptance.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {asInterceptor} from '@loopback/context';
+import {anOperationSpec} from '@loopback/openapi-spec-builder';
+import {get, param} from '@loopback/openapi-v3';
+import {
+  Client,
+  createRestAppClient,
+  expect,
+  givenHttpServerConfig,
+} from '@loopback/testlab';
+import {RestApplication} from '../../..';
+import {
+  cachedResults,
+  CachingInterceptorProvider,
+  clearCache,
+  status,
+} from './caching-interceptor';
+
+describe('global caching interceptor', () => {
+  let client: Client;
+  let app: RestApplication;
+
+  before(givenAClient);
+  after(async () => {
+    await app.stop();
+  });
+
+  context('toUpperCase', () => {
+    it('invokes the controller method if not cached', async () => {
+      await client.get('/toUpperCase/Hello').expect(200, 'HELLO');
+      expect(status.returnFromCache).to.be.false();
+    });
+
+    it('returns from cache without invoking the controller method', async () => {
+      for (let i = 0; i <= 5; i++) {
+        await client.get('/toUpperCase/Hello').expect(200, 'HELLO');
+        expect(status.returnFromCache).to.be.true();
+      }
+    });
+
+    it('invokes the controller method after cache is cleared', async () => {
+      clearCache();
+      await client.get('/toUpperCase/Hello').expect(200, 'HELLO');
+      expect(status.returnFromCache).to.be.false();
+    });
+  });
+
+  context('toLowerCase', () => {
```

Comment:
This is confusing. IIUC, this block is verifying that global interceptors are applied on route handlers.

Please rework `context` name to capture this information and move the definition of the route from the bottom of the test file into this particular `context` block that's using it.